Feature xuehui

renxuehui opened the pull request:
- 蓝绿拓扑图功能

renxuehui committed:
feat: 拓扑图功能

renxuehui committed:
Merge branch 'feature' of https://github.com/Nepxion/DiscoveryPlatform into feature

# Conflicts:
#	discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GrayDto.java
#	discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/service/GrayService.java
#	discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/service/GrayServiceImpl.java

HaojunRen committed:
Merge pull request #53 from Nepxion/feature

Merge

renxuehui committed:
fixed: 蓝绿拓扑图查看

renxuehui committed:
Merge branch 'feature' into feature-xuehui

renxuehui committed:
fixed: 补充作者信息

renxuehui committed:
fixed: 补充作者信息

renxuehui committed:
fixed: 作者信息

renxuehui committed:
Merge branch 'feature' into feature-xuehui

renxuehui committed:
fixed: 注释格式调整

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/tool/GraphUtils.java:
```diff
@@ -0,0 +1,96 @@
+		return nodeCache;
+	}
+
+	public static GraphDto changeRuleEntityToGraph(RuleEntity ruleEntity) {
```
change改成convert

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/tool/GraphUtils.java:
```diff
@@ -0,0 +1,96 @@
+
+public class GraphUtils {
+	private GraphUtils(){}
+	private static final String BEGIN_NODE_ID = "begin";
```
static final 一般放在类的第一行

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphNodeDto.java:
```diff
@@ -0,0 +1,70 @@
+	}
+
+	private String condition;
+	private String routeId;
```
移动到上面去

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphNodeDto.java:
```diff
@@ -0,0 +1,70 @@
+	private String id;
+	private String label;
+	private String type;
+	private String version;
```
version改成value

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphEdgeDto.java:
```diff
@@ -0,0 +1,42 @@
+
+	private String source;
+	private String target;
+	private String version;
```
去掉version

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphEdgeDto.java:
```diff
@@ -0,0 +1,42 @@
+ * @version 1.0
+ */
+
+public class GraphEdgeDto {
```
类命名成GraphLinkDto

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphDto.java:
```diff
@@ -0,0 +1,35 @@
+		return edges;
+	}
+
+	public void setEdges(List<GraphEdgeDto> edges) {
```
改成setLinks

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server-ui/src/main/resources/templates/blue-green/view.ftl:
```diff
@@ -0,0 +1,248 @@
+    var type = target.get('type');
+    if (type === 'circle') {
+      alert('你点击的是边上的圆点');
+    }
```
去掉alert

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/controller/BlueGreenPageController.java:
```diff
@@ -7,9 +7,12 @@
  * @version 1.0
  */
 
+import com.nepxion.discovery.common.util.JsonUtil;
```
导入模板，format import次序。下同

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphDto.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.List;
+
+public class GraphDto {
+
```
Dto集成序列化接口，序列化自动通过IDE生成，不要用1L表示

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphDto.java:
```diff
@@ -0,0 +1,35 @@
+
+	public void setEdges(List<GraphEdgeDto> edges) {
+		this.edges = edges;
+	}
```
增加如下方法，在底部
```
@Override 
public int hashCode() { 
return HashCodeBuilder.reflectionHashCode(this); 
} 
 @Override 
public boolean equals(Object object) { 
return EqualsBuilder.reflectionEquals(this, object); 
} 
 @Override 
public String toString() { 
return ToStringBuilder.reflectionToString(this, ToStringStyle.MULTI_LINE_STYLE); 
}
```

renxuehui committed:
fixed: review问题修改
- 代码格式模板导入
- 去掉edgeDto version； NodeDto, version => value
- hash、equals、toString;
- serialVersionUID;

renxuehui committed:
fixed: 拓扑图箭头样式调整

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server-ui/src/main/resources/templates/blue-green/list.ftl:
```diff
@@ -98,6 +98,13 @@
+                                <i class="layui-icon layui-icon-add-1"></i>&nbsp;&nbsp;查看<b>拓扑</b>图
+                            </button>
+                        </div>
+                        </@insert>
```
按钮风格需要调整一下

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphLinkDto.java:
```diff
@@ -0,0 +1,56 @@
+ * @author Xuehui Ren
+ * @version 1.0
+ */
+
```
把作者信息移到package和import之间

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphLinkDto.java:
```diff
@@ -0,0 +1,56 @@
+
+	private static final long serialVersionUID = 3536615004773370039L;
+	private String source;
+    private String target;
```
没有对齐？

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphNodeDto.java:
```diff
@@ -0,0 +1,92 @@
+ *
+ * @author Xuehui Ren
+ * @version 1.0
+ */
```
把作者信息移到package和import之间

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/entity/dto/GraphNodeDto.java:
```diff
@@ -0,0 +1,92 @@
+
+	private static final long serialVersionUID = 5728555476871710353L;
+	private String id;
+    private String label;
```
没有对齐？

HaojunRen commented on discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/service/BlueGreenService.java:
```diff
@@ -7,29 +7,33 @@
 import com.nepxion.discovery.platform.server.entity.po.BlueGreenPo;
 import com.nepxion.discovery.platform.server.service.base.BasePublishService;
+import java.util.Collection;
+import java.util.List;
```
import顺序format一下

renxuehui committed:
Merge branch 'feature' into feature-xuehui

renxuehui committed:
fixed: 拓扑图切换成图标模式

renxuehui committed:
fixed: 增加首节点显示文字；弹窗底部按钮改成【关闭】；

HaojunRen committed:
Modify Button Style

HaojunRen committed:
Modify Dialog style

HaojunRen committed:
Modify Link UI

renxuehui committed:
fixed: 1. api网关名称、服务名称 居中； 2. 蓝绿颜色交替显示；

renxuehui committed:
fixed: 代码review问题

renxuehui committed:
fixed:  deal code review
- UI蓝绿颜色交替显示;
- 选中记录才能查看拓扑图;
- reformat for tab to 4 spaces;
- 去掉后台测试代码，改成正式： RuleEntity ruleEntity = platformDiscoveryAdapter.getConfig(portalName)

renxuehui committed:
Merge branch 'feature' into feature-xuehui

# Conflicts:
#	discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/service/BlueGreenService.java
#	discovery-platform-server/discovery-platform-starter-server/src/main/java/com/nepxion/discovery/platform/server/service/BlueGreenServiceImpl.java

renxuehui committed:
fixed: 删除无用代码

HaojunRen committed:
Fix coding style

HaojunRen committed:
Refactory GraphEdgeDto

HaojunRen committed:
Refactory GraphUtils

HaojunRen committed:
Refactory GraphTool

renxuehui committed:
fixed: 未发布版本，查看拓扑图效果; GraphTool重构=》 BlueGreenAndGraphTool

renxuehui committed:
fixed: 类名修改; debugger去除;

renxuehui committed:
fixed: 查看拓扑图勾选多个时，js异常导致未提示

renxuehui committed:
fixed: 拓扑图支持节点拖动，caven模式更顺滑
